File: danceconnect.client/src/app/views/setup/instructor-profile/instructor.ts
export interface Instructor {
  name: string;
  gender?: string;
  dob: Date;
  phone: string;
  hourlyRate: number;
  profilePic?: File;
  identityDocument?: File;
  shortIntroVideo?: File;
  street: string;
  city?: string;
  province?: string;
  postalCode?: string;
}

export interface InstructorResponse {
  id: number;
  name: string;
  gender?: string;
  dob: string;
  phone: string;
  email: string;
  averageRating: number;
  profileStatus: string;
  hourlyRate: string;
  profilePic?: string;
  identityDocument?: string;
  shortIntroVideo?: string;
  street: string;
  city?: string;
  province?: string;
  postalCode?: string;
}

File: DanceConnect.Server/Entities/Instructor.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using DanceConnect.Server.Enums;

namespace DanceConnect.Server.Entities
{
    public class Instructor
    {
        [Key]
        public int InstructorId { get; set; }

        [Required, MinLength(10)]
        public string? Name { get; set; }

        [Required]
        public string? Gender { get; set; }
        [Required]
        public DateTime Dob { get; set; }

        [Required]
        [RegularExpression(@"^\d{10}$")]
        public string? Phone { get; set; }

        public decimal HourlyRate { get; set; }
        public ProfileStatus ProfileStatus { get; set; }

        public string? ProfilePic { get; set; }

        public string? IdentityDocument { get; set; }

        public string? IntroVideo { get; set; }

        [Required]
        public string? Street { get; set; }

        [Required]
        public string? City { get; set; }

        [Required]
        [RegularExpression(@"^\d+$")]
        public string? PostalCode { get; set; }

        [Required]
        public string? Province { get; set; }

        [ForeignKey(nameof(AppUser))]
        public int AppUserId { get; set; }
        public ApplicationUser? AppUser { get; set; }

        public ICollection<Rating> Ratings { get; set; } = [];
    }
}
